### PR TITLE
Fixing some more test issues

### LINKED ISSUE #241: Flacky test TestStreamConnection::connectionUpDown
One test occasionally fails for me: TestStreamConnection::connectionUpDown

##Failure Location unknown## : Error
Test name: TestStreamConnection::connectionUpDown
uncaught exception of type std::exception (or derived).
- Search permission is denied on a component of the path prefix.

Other runs it's OK so it's hard to tell, can be a race condition. Logging to look into this later.

### PR DESCRIPTION
**Pull request by dklimkin**
Fixing more test issues with running test on my platform. See line level comments for specific details.
Fixes #241 .

### CODE REVIEW

File: ibrcommon/tests/link/netlinktest.cpp [hunk 1]
```diff
@@ -26,20 +26,21 @@ void netlinktest :: baseTest (void)
...
-		std::cout << " " << (*iter).toString() << std::endl;
+	try {
+		std::list<ibrcommon::vaddress> ret = ibrcommon::LinkManager::getInstance().getAddressList(iface);
+		std::cout << "Addresses:" << std::endl;
+		for (std::list<ibrcommon::vaddress>::iterator iter = ret.begin(); iter != ret.end(); ++iter)
+		{
+			std::cout << " " << (*iter).toString() << std::endl;
+		}
+	} catch(const ibrcommon::Exception &e) {
+		std::cout << "\tInterface enumeration is not allowed on this platform: "
```
Comment (dklimkin) [inline ibrcommon/tests/link/netlinktest.cpp]: Changes in this block should be noop, except it's more informative for test, and also gives me space to put ifdefs for my platform.

File: ibrcommon/tests/net/tcpstreamtest.cpp [hunk 2]
```diff
@@ -35,20 +35,23 @@ void tcpstreamtest :: tearDown (void)
 
 void tcpstreamtest :: baseTest (void)
 {
-	StreamChecker _checker(4343, 10);
+	const int _testPort = 4343;
```
Comment (dklimkin) [inline ibrcommon/tests/net/tcpstreamtest.cpp]: Extracting so a port can be replaced for substitution by a test system.

File: ibrcommon/tests/net/tcpstreamtest.cpp [hunk 3]
```diff
@@ -89,8 +91,10 @@ void tcpstreamtest::runTest()
 	} catch (const ibrcommon::vsocket_interrupt &e) {
 		// select interrupted
 	} catch (const ibrcommon::socket_exception &e) {
-		CPPUNIT_FAIL(std::string("client error: ") + e.what());
+		std::cerr << "client error: " << + e.what() << std::endl;
```
Comment (dklimkin) [inline ibrcommon/tests/net/tcpstreamtest.cpp]: CPPUNIT_FAIL here results in main thread termination but listening thread to keep waiting unjoined. To if the client connection fails, the test hangs indefinitely. With this change it fails fast.

File: ibrcommon/tests/net/tcpstreamtest.h [hunk 4]
```diff
@@ -60,7 +60,7 @@ class tcpstreamtest : public CPPUNIT_NS :: TestFixture
 
 	protected:
 		void baseTest (void);
-		void runTest (void);
+		bool runTest (const int port);
```
Comment (dklimkin) [inline ibrcommon/tests/net/tcpstreamtest.h]: Just to be able to specify the port number in one place.

File: ibrcommon/tests/unittests/Base64StreamTest.cpp [hunk 5]
```diff
@@ -64,19 +66,15 @@ void Base64StreamTest::testFileReference()
 	std::stringstream ss_decoded;
 
 	// open reference file
-	std::fstream f("../base64-enc.dat");
-
-	CPPUNIT_ASSERT(f.good());
+	std::string filename("../base64-enc.dat");
+	std::ifstream f(filename);
```
Comment (dklimkin) [inline ibrcommon/tests/unittests/Base64StreamTest.cpp]: fstream() fails if the file is read-only. Changing to ifstream.

File: ibrcommon/tests/unittests/Base64StreamTest.cpp [hunk 6]
```diff
@@ -64,19 +66,15 @@ void Base64StreamTest::testFileReference()
...
 	ss_decoded << ibrcommon::Base64Reader(f, 46207).rdbuf() << std::flush;
+	CPPUNIT_ASSERT_MESSAGE("Error in data decode", f.good());
 
-	CPPUNIT_ASSERT(f.good());
-
-//	std::cout << std::endl;
-//	std::cout << ss_decoded.rdbuf() << std::endl;
-
-	std::fstream fd("../base64-dec.dat");
+	std::ifstream fd("../base64-dec.dat");
```
Comment (dklimkin) [inline ibrcommon/tests/unittests/Base64StreamTest.cpp]: The same.

File: ibrdtn/ibrdtn/ibrdtn/data/CompressedPayloadBlock.cpp [hunk 7]
```diff
@@ -165,8 +167,8 @@ namespace dtn
 
 					int ret, flush;
 					uInt have;
-					unsigned char in[CHUNK_SIZE];
-					unsigned char out[CHUNK_SIZE];
+					std::vector<unsigned char> in(CHUNK_SIZE);
```
Comment (dklimkin) [inline ibrdtn/ibrdtn/ibrdtn/data/CompressedPayloadBlock.cpp]: Same as before -- to much on the stack frame.

File: ibrdtn/ibrdtn/tests/net/TestStreamConnection.cpp [hunk 8]
```diff
@@ -217,16 +217,15 @@ void TestStreamConnection::connectionUpDown()
...
-	// create a new server bound to tcp port 1234
-	testserver srv(new ibrcommon::tcpserversocket(1234));
+	// create a new server bound to a tcp port.
+	testserver srv(new ibrcommon::tcpserversocket(_testPort));
 
 	// start the server thread
 	srv.start();
 
-	ibrcommon::vaddress addr("127.0.0.1", 1234);
+	ibrcommon::vaddress addr("localhost", _testPort);
```
Comment (dklimkin) [inline ibrdtn/ibrdtn/tests/net/TestStreamConnection.cpp]: "127.0.0.1" is ipv4 only, fails on IPv6-only systems. Localhost should work on both.

File: ibrcommon/tests/net/tcpstreamtest.cpp [hunk 9]
```diff
@@ -64,18 +67,17 @@ tcpstreamtest::StreamChecker::~StreamChecker()
 	_sock.destroy();
 }
 
-void tcpstreamtest::runTest()
+bool tcpstreamtest::runTest(const int port)
 {
 	char values[10] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' };
 
 	try {
-		ibrcommon::vaddress addr("127.0.0.1", 4343);
```
Comment (dklimkin) [inline ibrcommon/tests/net/tcpstreamtest.cpp]: Fails on IPv6-only systems.

**Comment by dklimkin** [comment]
Ha! looks like some C+11 validation works after all!

**Comment by dklimkin** [comment]
Friendly ping on this one.

**Comment by morgenroth** [comment]
I have noticed it. I will review the code once I have time.

**Comment by dklimkin** [comment]
Closing due to inactivity, will keep patching in a fork.

**Pull request closed without merging**